service/controller/resource/namespace/current.go

```diff
@@ -71,12 +74,17 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 	// remove the namespace as usual.
 	if key.IsDeleted(&customObject) {
 		n := key.ClusterNamespace(customObject)
-		list, err := r.k8sClient.CoreV1().Pods(n).List(ctx, metav1.ListOptions{})
+
+		lo := metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", label.ManagedBy, project.Name()),
```

Comment:
```suggestion
			LabelSelector: fmt.Sprintf("%s=%s", label.ManagedBy, key.OperatorName),
```
`project.Name()` and `key.OperatorName` should both be `kvm-operator`, but this just makes sure that we're following the deployment label definition here https://github.com/giantswarm/kvm-operator/blob/v3.14.2/service/controller/resource/deployment/master_deployment.go#L93.

Reply:
Thanks for pointing it out!